web.rb

```diff
@@ -57,9 +60,84 @@
     "Charge declined"
   end
 
+end 
+
+#endpoint for Stripe Connect Oauth
+get '/connect' do 
+  #gets two params from Stripe OAUTH
+  scope = params[:scope]
+  auth_code = params[:code]
+
+  p scope
+  p auth_code
+
+  #creates account with Stripe or returns error to Sinatra 
+
+  begin
+    response = RestClient.post 'https://connect.stripe.com/oauth/token', {
+      client_secret: $PRIVATE_TEST_KEY,
+      code: auth_code, 
+      grant_type: 'authorization_code'}
+ 
+
+    response = JSON.parse(response.to_s)
+
+    db = SQLite3::Database.new("stripe_store.db")
+    db.execute("INSERT INTO accounts (
+      authorization_code,
+      token_type,
+      stripe_publishable_key,
+      scope,
+      livemode,
+      stripe_user_id,
+      refresh_token,
+      access_token) VALUES (?, ?, ?, ?, ?, ?, ?, ?)", [
+      auth_code, 
+      response['token_type'], 
+      response['stripe_publishable_key'], 
+      response['scope'],
+      response['livemode'].to_s,
+      response['stripe_user_id'],
+      response['refresh_token'],
+      response['access_token']
+      ])
+    
+    puts "Account #{response['stripe_user_id']} created." 
+
+    redirect "/account_confirmation?acct_id=#{response['stripe_user_id']}"
+
+  #error handling for restclient and other errors (not really)
```

Comment:
You can handle errors if you want, but usually you'll drive yourself nuts trying to. It's more typical to only handle EXPECTED errors. So for instance, if form data were validated on the backend, it might be expected that sometimes the user fills in bad data in a form, and an error is thrown. Catching that error would make sense; that errors is part of the everyday routine of doing business.